test_suite.py

```diff
@@ -18,6 +19,31 @@ def expected_read_file_output():
     return expected
 
 
+class Language(ABC):
+    name: str
+
+    @abstractmethod
+    def script_command_parts(self, test_name):
+        pass
+
+    def script(self, test_name):
+        return " ".join(self.script_command_parts(test_name))
```

Comment:
Is it weird to have a concrete method call an abstract method? This seems to work and this way the classes don't have to implement it.

Reply:
Not weird at all. I think this is called the "hook" pattern.